device/core/test/_client_test.js

```diff
@@ -96,9 +67,27 @@ describe('Client', function () {
       clock.restore();
       testCallback();
     });
+
+    /*Tests_SRS_NODE_DEVICE_CLIENT_16_087: [The `fromConnectionString` method shall create a new `SharedAccessKeyAuthorizationProvider` object with the connection string passed as argument if it contains a SharedAccessKey parameter and pass this object to the transport constructor. ]*/
+    it('creates a SharedAccessKeyAuthorizationProvider and passes it to the transport', function (testCallback) {
+      var client = Client.fromConnectionString(sharedKeyConnectionString, function (authProvider) {
```

Comment:
>function (authProvider) [](start = 74, length = 23)

nit: I'm not sure what I think about this.  It feels dirty.  I know it's perfectly legal to pass a callback as a constructor function, but there's a conceptual difference here that makes me uneasy.  Or maybe not.  I'm not sure.

Reply:
it's not a callback, it's just a function. just like a constructor. i could define a constructor outside, call it constructor, and then pass it here, but that'd be exactly the same thing

---
In reply to: [157075622](https://github.com/Azure/azure-iot-sdk-node/pull/180#discussion_r157075622) [](ancestors = 157075622)